src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -528,17 +534,25 @@ public override Symbol VisitDynamicType(DynamicTypeSymbol symbol)
                 switch (otherContainer.Kind)
                 {
                     case SymbolKind.Namespace:
-                        if (sourceType is AnonymousTypeManager.AnonymousTypeTemplateSymbol template)
+                        if (sourceType is AnonymousTypeManager.AnonymousTypeTemplateSymbol typeTemplate)
                         {
                             Debug.Assert((object)otherContainer == (object)_otherAssembly.GlobalNamespace);
-                            TryFindAnonymousType(template, out var value);
+                            TryFindAnonymousType(typeTemplate, out var value);
                             return (NamedTypeSymbol?)value.Type?.GetInternalSymbol();
                         }
-                        else if (sourceType is SynthesizedDelegateSymbol delegateSymbol)
+                        else if (sourceType is AnonymousTypeManager.AnonymousDelegateTemplateSymbol delegateTemplate)
                         {
                             Debug.Assert((object)otherContainer == (object)_otherAssembly.GlobalNamespace);
-                            TryFindSynthesizedDelegate(delegateSymbol, out var value);
-                            return (NamedTypeSymbol?)value.Delegate?.GetInternalSymbol();
+                            if (delegateTemplate.HasFixedTypes)
+                            {
+                                TryFindAnonymousDelegateWithFixedType(delegateTemplate, out var value);
```

Comment:
>TryFindAnonymousDelegateWithFixedType [](http://example.com/codeflow?start=32&length=37)

Was the logic backwards before? It looks like the condition didn't change its meaning, but the branches got flipped.

Reply:
The branches were flipped because `HasFixedTypes == !IsParameterizedDelegateType`.